diff --git a/drivers/hid/hid-multitouch.c b/drivers/hid/hid-multitouch.c
--- a/drivers/hid/hid-multitouch.c
+++ b/drivers/hid/hid-multitouch.c
@@ -453,7 +453,6 @@
 	int ret;
 	u32 size = hid_report_len(report);
 	u8 *buf;
-	struct hid_report_enum *rep_enum;

 	/*
 	 * Do not fetch the feature report if the device has been explicitly
diff --git a/fs/f2fs/segment.c b/fs/f2fs/segment.c
--- a/fs/f2fs/segment.c
+++ b/fs/f2fs/segment.c
@@ -3897,7 +3897,7 @@
 	/* change the current segment */
 	if (segno != curseg->segno) {
 		curseg->next_segno = segno;
-		change_curseg(sbi, type, true);
+		change_curseg(sbi, type);
 	}

 	curseg->next_blkoff = GET_BLKOFF_FROM_SEG0(sbi, blkaddr);
@@ -3906,7 +3906,7 @@
 	/* restore current segment */
 	if (old_cursegno != curseg->segno) {
 		curseg->next_segno = old_cursegno;
-		change_curseg(sbi, type, true);
+		change_curseg(sbi, type);
 	}
 	curseg->next_blkoff = old_blkoff;

diff --git a/drivers/md/dm-verity-debug.c b/drivers/md/dm-verity-debug.c
--- a/drivers/md/dm-verity-debug.c
+++ b/drivers/md/dm-verity-debug.c
@@ -255,6 +255,7 @@
     u8 *page;

     int i;
+    int ce;
     char hex_str[65] = {0, };
     /* Corruption should be visible in device status in all modes */
     v->hash_failed = 1;
@@ -271,10 +272,11 @@
     }
 #endif

-    if (v->corrupted_errs >= DM_VERITY_MAX_CORRUPTED_ERRS)
-        goto out;
-
-    v->corrupted_errs++;
+    ce = atomic_read(&v->corrupted_errs);
+    do {
+        if (ce >= DM_VERITY_MAX_CORRUPTED_ERRS)
+            goto out;
+    } while (!atomic_try_cmpxchg(&v->corrupted_errs, &ce, ce + 1));

     switch (type) {
         case DM_VERITY_BLOCK_TYPE_DATA:
@@ -320,7 +322,7 @@

     panic("dmv corrupt");

-    if (v->corrupted_errs == DM_VERITY_MAX_CORRUPTED_ERRS)
+    if (ce + 1 == DM_VERITY_MAX_CORRUPTED_ERRS)
         DMERR("%s: reached maximum errors", v->data_dev->name);

     snprintf(verity_env, DM_VERITY_ENV_LENGTH, "%s=%d,%llu",
