diff --git a/kernel/supercall/supercall.c b/kernel/supercall/supercall.c
--- a/kernel/supercall/supercall.c
+++ b/kernel/supercall/supercall.c
@@ -358,15 +358,21 @@ void __init ksu_supercalls_init(void)
 
     ksu_supercall_dump_commands();
 
+#ifndef CONFIG_KSU_SUSFS
     rc = register_kprobe(&reboot_kp);
     if (rc) {
         pr_err("reboot kprobe failed: %d\n", rc);
     } else {
         pr_info("reboot kprobe registered successfully\n");
     }
+#endif // #ifndef CONFIG_KSU_SUSFS
 }
 
 void __exit ksu_supercalls_exit(void)
 {
+#ifndef CONFIG_KSU_SUSFS
     unregister_kprobe(&reboot_kp);
+#else
+    pr_info("susfs: do nothing\n");
+#endif // #ifndef CONFIG_KSU_SUSFS
     ksu_supercall_cleanup_state();
 }
